### PR TITLE
Add my patches, fixing yatoclip, speed up yatoclip build by 30-50%

### PR DESCRIPTION
**Pull request by TheArcaneBrony**
- Modified eula.txt because awoo
- Enabled incremental builds (30-50% speedup, went from 2m50 to 1m40)
- Backported yatoclip fix from c2me branch
- Added permission yatopia.plugins to /pl and /plugins, and yatopia.version to /version
- Cleaned up some newlines

### CODE REVIEW

**Review by josephworks — APPROVED** [review]

**Review by josephworks — CHANGES_REQUESTED** [review]
Could you set the JVM args to either ``-Xms2G -Xmx2G`` or ``-Xms4G -Xmx4G`` in the gradle.properties?
Also, add a new line at end of the file.

**Review by josephworks — APPROVED** [review]

**Comment by TheArcaneBrony** [comment]
Changed the JVM args to -Xmx2G -Xmx2G in gradle.properties and added newline, as requested

**Review by josephworks — DISMISSED** [review]
Patch 69, the patch that modifies the EULA, noone knows who Momiji is and he isn't a community staple.
I love the commit, but I don't think it is right for YatopiaMC.

For those wondering:
image

P.S. I am more of an anime girl lover imo yes. ZeroTwo, Enterprise, etc...

**Comment by TheArcaneBrony** [comment]
Momiji Inubashiri is who I meant in that commit.
See: https://knowyourmeme.com/memes/awoo

**Comment by TheArcaneBrony** [comment]
Dropped patch 70 as it doesn't serve any purpose aside code aesthetics, and may slightly increase memory usage due to wildcard importing.

**Comment by Kylepoops** [comment]
> Momiji Inubashiri is who I meant in that commit.
> See: https://knowyourmeme.com/memes/awoo

犬走椛 wwww

**Comment by duplexsystem** [comment]
Do Hatsune Miku, the creator of Minecraft, for the EULA

**Comment by Titaniumtown** [comment]
wut

File: patches/server/0069-Momiji-best-awoo.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+--- a/src/main/java/net/minecraft/server/EULA.java
++++ b/src/main/java/net/minecraft/server/EULA.java
+@@ -83,9 +83,8 @@ public class EULA {
+ 
+                 try {
+                     Properties properties = new Properties();
+-
+                     properties.setProperty("eula", "false");
+-                    properties.store(outputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula).\nYou also agree that tacos are tasty, and the best food in the world.");  // Paper - fix lag; // Tuinity - Tacos are disgusting // Purpur - no they're not
++                    properties.store(outputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula).\nYou also agree that tacos are tasty, and the best food in the world.\nIn addition, you also agree that Momiji Inubashiri is best awoo!");  // Paper - fix lag; // Tuinity - Tacos are disgusting // Purpur - no they're not // TheArcaneBrony @ Yatopia - Momiji best awoo! AWOOOOO!~
```
Comment (Titaniumtown) [inline patches/server/0069-Momiji-best-awoo.patch]: :gun: no (I don't even know what this means)

**Comment by Titaniumtown** [comment]
You need to update the jenkinsfile too

**Comment by josephworks** [comment]
> Do Hatsune Miku, the creator of Minecraft, for the EULA

@duplexsystem  https://youtu.be/qh-EqX6pSzQ

**Comment by TheArcaneBrony** [comment]
what the... why

**Comment by Titaniumtown** [comment]
What has this PR's comments become 😆

**Comment by TheArcaneBrony** [comment]
no idea, pls accept pr :pray:

File: patches/server/0069-Momiji-best-awoo.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+index b6d94005cb2d7f28fc4f0b78191ef1ba13190e56..af13f50817f5c701c4e7331711580e5c999e5219 100644
+--- a/src/main/java/net/minecraft/server/EULA.java
++++ b/src/main/java/net/minecraft/server/EULA.java
+@@ -83,9 +83,8 @@ public class EULA {
+ 
+                 try {
+                     Properties properties = new Properties();
+-
+                     properties.setProperty("eula", "false");
+-                    properties.store(outputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula).\nYou also agree that tacos are tasty, and the best food in the world.");  // Paper - fix lag; // Tuinity - Tacos are disgusting // Purpur - no they're not
```
Comment (Titaniumtown) [inline patches/server/0069-Momiji-best-awoo.patch]: Drop this then I'll accept the PR

**Comment by ChrisChrome** [comment]
awoo

**Review by Titaniumtown — CHANGES_REQUESTED** [review]

File: patches/server/0069-Momiji-best-awoo.patch (hunk 2 above)
Comment (ChrisChrome) [reply patches/server/0069-Momiji-best-awoo.patch]: boo (arcane made me)

Comment (TheArcaneBrony) [reply patches/server/0069-Momiji-best-awoo.patch]: cmon jerry lmao

Comment (TheArcaneBrony) [reply patches/server/0069-Momiji-best-awoo.patch]: gotta have dat awoo\:tm:

Comment (Titaniumtown) [reply patches/server/0069-Momiji-best-awoo.patch]: After @TheArcaneBrony explained the meaning of "awoo" privately, I don't see the harm in it being added (even though it's a bit strange to me).

**Review by Titaniumtown — APPROVED** [review]

**Comment by Titaniumtown** [comment]
Testing is needed on newer versions of java, once that has happened, I'll merge.

**Comment by TheArcaneBrony** [comment]
Just tested Yatoclip on java 17, worked fine.
```
thearcanebrony@DESKTOP-0ABLLGL:~/test$ ~/jdk-17/bin/java -DPaper.IgnoreJavaVersion=true -jar Yatoclip-1.16.5-R0.1-SNAPSHOT-all.jar
Downloading version.json...
Downloading BuildData...
Downloading vanilla jar...
Setting up SpecialSource: cache/1.16.5/BuildData/bin/SpecialSource-2.jar
Applying class mapping...
Invoking SpecialSource with arguments: [map, --only, ., --only, net/minecraft, --only, com/mojang/math, --auto-lvt, BASIC, --auto-member, SYNTHETIC, -i, /home/thearcanebrony/test/cache/1.16.5/Minecraft/1.16.5.jar, -m, /home/thearcanebrony/test/cache/1.16.5/BuildData/mappings/bukkit-1.16.5-cl.csrg, -o, /home/thearcanebrony/test/cache/1.16.5/Minecraft/1.16.5-cl.jar]

Applying member mapping...
Invoking SpecialSource with arguments: [map, --only, ., --only, net/minecraft, --only, com/mojang/math, --auto-member, LOGGER, --auto-member, TOKENS, -i, /home/thearcanebrony/test/cache/1.16.5/Minecraft/1.16.5-cl.jar, -m, /home/thearcanebrony/test/cache/1.16.5/BuildData/mappings/bukkit-1.16.5-members.csrg, -o, /home/thearcanebrony/test/cache/1.16.5/Minecraft/1.16.5-m.jar]

Releasing SpecialSource
Applying patches...
Yatoclip server setup completed in 3319.84ms
Unsupported Java detected (61.0). Only up to Java 16 is supported.
System Info: Java 17 (OpenJDK 64-Bit Server VM 17-ea+23-2064) Host: Linux 5.10.16.3-microsoft-standard-WSL2 (amd64)
Loading libraries, please wait...
[18:55:14 INFO]: Finished building data fixers after 3ms
[18:55:15 ERROR]: Failed to load properties from file: server.properties
[18:55:15 INFO]: [STDOUT]: No global eula found at /home/thearcanebrony/eula.txt
[18:55:15 WARN]: Failed to load eula.txt
[18:55:15 INFO]: You need to agree to the EULA in order to run the server. Go to eula.txt for more info.```

**Comment by TheArcaneBrony** [comment]
Is this still open? I thought this was fine to be merged?

File: build.gradle.kts [hunk 3]
```diff
@@ -66,4 +66,7 @@ subprojects {
         targetCompatibility = JavaVersion.current()
         withSourcesJar()
     }
+    tasks.withType<JavaCompile>().configureEach {
+        options.isIncremental = true
```
Comment (Toffikk) [inline build.gradle.kts:70]:
```suggestion
        options.isIncremental = true
        options.isFork = true
        options.encoding = "UTF-8"

Comment (TheArcaneBrony) [reply build.gradle.kts:70]: makes sense? though wouldnt you want ASCII encoding?

**Comment by MrIvanPlays** [comment]
yatoclip kekw

**Comment by TheArcaneBrony** [comment]
yes, yatoclip fast development™️

**Comment by MrIvanPlays** [comment]
> yes, yatoclip fast development™️

the funniest joke i've ever heard in my life. I cannot stop laughing

**Comment by TheArcaneBrony** [comment]
i mean for some reason its faster then using paperclip in my experience

**Comment by TheArcaneBrony** [comment]
Ffs @Titaniumtown

**Comment by Titaniumtown** [comment]
@MrIvanPlays This is a place to be constructive and aid in development, if you're just going to use this as a place to make fun of things without contributing to the subject at hand. This isn't the place for you.

**Comment by TheArcaneBrony** [comment]
thats not what i meant @Titaniumtown lol,  i meant you made changes that somehow conflict with my PR 
tho i don't see the issue with yatoclip, its faster IN MY EXPERIENCE, doesn't speak for everyone obviously.

**Comment by Toffikk** [comment]
@MrIvanPlays yatoclip is good lol

**Comment by jokbon** [comment]
sorry but i have to agree with ivan that yatoclip is a unfunny joke and needs improvements

**Comment by tommasobenatti** [comment]
> sorry but i have to agree with ivan that yatoclip is a unfunny joke and needs improvements

What issues did you encounter?

**Comment by Toffikk** [comment]
it needs improvements for sure, thats why its still in development and it needs some bug-fixes but it works and it does its job pretty good

**Comment by TheArcaneBrony** [comment]
I personally havent had issues with yatoclip aside on windows, so like, its fine i guess

**Comment by Toffikk** [comment]
yatoclip doesnt like windows lol

```windows
Applying patches...
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.util.zip.ZipException: duplicate entry: net/minecraft/server/v1_16_R3/Tags$1.class
        at org.yatopiamc.yatoclip.YatoclipPatcher.patchJar(YatoclipPatcher.java:151)
        at org.yatopiamc.yatoclip.ServerSetup.applyMappingsAndPatches(ServerSetup.java:92)
        at org.yatopiamc.yatoclip.ServerSetup.setup(ServerSetup.java:50)
        at org.yatopiamc.yatoclip.Yatoclip.main(Yatoclip.java:18)
Caused by: java.lang.RuntimeException: java.util.zip.ZipException: duplicate entry: net/minecraft/server/v1_16_R3/Tags$1.class
        at org.yatopiamc.yatoclip.YatoclipPatcher.patchJar(YatoclipPatcher.java:147)
        ... 3 more
Caused by: java.util.zip.ZipException: duplicate entry: net/minecraft/server/v1_16_R3/Tags$1.class
        at java.base/java.util.zip.ZipOutputStream.putNextEntry(ZipOutputStream.java:233)
        at org.yatopiamc.yatoclip.YatoclipPatcher.putNextEntrySafe(YatoclipPatcher.java:195)
        at org.yatopiamc.yatoclip.YatoclipPatcher.patchJar(YatoclipPatcher.java:126)
        ... 3 more
        ```

**Review by Toffikk — APPROVED** [review]

**Pull request closed without merging**